### DISCUSSION TITLE
A better way to enable add-on Terraform for different cloud providers?

### DISCUSSION THREAD

**Discussion by zzxwill** [kubevela/kubevela#2173] [Ideas] [answered]
`vela addon enable terraform <CloudProvider> -<CredentialKey>=<CredentialValue> ...`

By default, <CloudProvider> is `alibaba`, without setting it, terraform controller chart will be installed. A secret and a Provider object will be created.

Then if the platform team wants to provision AWS, he/she runs the command.

`vela addon enable terraform was -AWS_ACCESS_KEY_ID=xxx -AWS_SECRET_ACCESS_KEY=yyy`. 

The terraform controller will NOT be installed again. Only the provider for AWS will created.

**Comment by wonderflow** [comment, marked as answer]
```
vela secret my-sec  AWS_ACCESS_KEY_ID=xxx -AWS_SECRET_ACCESS_KEY=yyy
vela addon enable terraform-provider-aws --secret my-sec
```

**Reply by zzxwill** [reply]
The better solution from @wonderflow
`vela addon enable terraform-provider-aws --AWS_ACCESS_KEY_ID=xxx -AWS_SECRET_ACCESS_KEY=yyy`

**Reply by chivalryq** [reply]
> vela addon enable terraform-provider-aws --secret 'AWS_ACCESS_KEY_ID=xxx -AWS_SECRET_ACCESS_KEY=yyy'

I suggest `vela addon enable terraform/provider-alicloud`, which indicates the hierarchical relationship between terraform and provider addons.

It would be an logical represent and implemented by replace “/” with “-”.

**Reply by wonderflow** [reply]
@chivalryq  cool, it's good

**Reply by zzxwill** [reply]
Great, let's use `vela addon enable terraform/provider-aws --AWS_ACCESS_KEY_ID=xxx -AWS_SECRET_ACCESS_KEY=yyy`
